reactnative/src/components/screens/auth/LoginScreen/LoginScreen.tsx

```diff
@@ -0,0 +1,75 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import React, { useContext, useState } from 'react'
+import { Box, Button, Center, Flex, Heading } from 'native-base'
+import { Field, Formik, FormikHelpers } from 'formik'
+import { NativeStackNavigationProp } from '@react-navigation/native-stack'
+import { ParamListBase } from '@react-navigation/native'
+
+import { api } from 'api'
+import { initialValues } from './loginForm'
+import { FormikInput } from 'components/form/FormikInput'
+import { LoginForm } from 'common-ts/dist/webServiceApi/interface/auth'
+import { UserContext } from 'components/context/UserContext'
+
+interface Props {
+  navigation: NativeStackNavigationProp<ParamListBase>
+}
+
+export const LoginScreen: React.FC<Props> = ({ navigation }) => {
+  const { setToken } = useContext(UserContext)
+  const [loading, setLoading] = useState(false)
+
+  const onSuccess = async (token: string) => {
+    await AsyncStorage.setItem('token', token)
+    setToken(token)
+    navigation.navigate('Keys')
+  }
+
+  const onFailure = ({ setFieldError }: FormikHelpers<LoginForm>) => {
+    setFieldError('email', 'Invalid credentials')
+  }
+
+  const onUnauthorized = ({ setFieldError }: FormikHelpers<LoginForm>) => {
+    setFieldError('email', 'This app is for admins only')
+  }
+
+  const onSubmit = async (
+    values: LoginForm,
+    helpers: FormikHelpers<LoginForm>,
+  ) => {
+    setLoading(true)
+    api.auth.login(values).then(({ data }) => {
+      setLoading(false)
+      if (data?.role === 'ROLE_ADMIN') return onSuccess(data.token)
+
```

Comment:
troche słaba obsługa wyjątków jest, bo ciągle dostaje jakieś promise rejection jak coś źle zrobie

Reply:
promise rejection z założenia nie powinno występować, gdy jest odpalony backend

można zrobić ogólną obsługę, że gdy serwer nie odpowiada to jest alert('Network error') ale to by byla jedna implementacja w api.ts

Reply:
dla mnie git, lepsze to niż promise rejection, który użytkownikowi nic nie mówi